### PR TITLE
Pcap conditional v2.2.10

### PR DESCRIPTION
**Pull request by scottfgjordan**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- Cleaned up PR based on feedback in https://github.com/OISF/suricata/pull/6473
- Corrected commit history so that code fixes weren't separate commits 
- Still TODO: If desired I can implement a version with head room for the TCPSegment storage instead. I wanted to get this PR out for review first and discuss here if that's what we want to do.

suricata-verify-pr: 542

### CODE REVIEW

File: doc/userguide/configuration/suricata-yaml.rst [hunk 1]
```diff
@@ -403,6 +403,16 @@ By default all packets are logged except:
 - TCP streams beyond stream.reassembly.depth
 - encrypted streams after the key exchange
 
+It is possible to do conditional pcap logging by using the `conditional`
+option in the pcap-log section. By default the variable is set to `all`
+so all packet are logged. If the variable is set to `alerts` then only
```
Comment (jufajardini) [inline doc/userguide/configuration/suricata-yaml.rst:408]: nit: packets

File: doc/userguide/configuration/suricata-yaml.rst [hunk 2]
```diff
@@ -403,6 +403,16 @@ By default all packets are logged except:
...
+the flow with alerts will be logged. If the variable is set to `tag`
+then only packets tagged by signature using the `tag` keyword will
```
Comment (jufajardini) [inline doc/userguide/configuration/suricata-yaml.rst:410]: nit: signatures

File: src/log-pcap.c [hunk 3]
```diff
@@ -574,37 +610,17 @@ static int PcapLog (ThreadVars *t, void *thread_data, const Packet *p)
...
-        if (out_size > 0) {
-            pl->size_current += out_size;
-            comp->bytes_in_block = len;
-        } else {
-            comp->bytes_in_block += len;
-        }
+    ret = PcapWrite(pl, comp, GET_PKT_DATA(p), len);
+#else
+    ret = PcapWrite(pl, NULL, GET_PKT_DATA(p), len);
+#endif
```
Comment (jufajardini) [inline src/log-pcap.c:697]: nit: readd `/* HAVE_LIBLZ4 */` ?

**Comment by scottfgjordan** [comment]
Sorry about the force push, accidentally fetched and merged upstream for this branch on the web UI. Force pushed to get rid of merge commit. No changes were made save for a rebase.

**Comment by scottfgjordan** [comment]
Replaced by: https://github.com/OISF/suricata/pull/6723

**Pull request closed without merging**